src/crimson/admin/osd_admin.cc

```diff
@@ -226,22 +226,35 @@ class DumpMetricsHook : public AdminSocketHook {
                                       std::string_view format,
                                       ceph::bufferlist&& input) const final
   {
-    std::unique_ptr<Formatter> f{Formatter::create(format, "json-pretty", "json-pretty")};
+    std::unique_ptr<Formatter> fref{Formatter::create(format, "json-pretty", "json-pretty")};
+    auto *f = fref.get();
     std::string prefix;
     cmd_getval(cmdmap, "group", prefix);
     f->open_object_section("metrics");
-    for (const auto& [full_name, metric_family]: seastar::scollectd::get_value_map()) {
-      if (!prefix.empty() && full_name.compare(0, prefix.size(), prefix) != 0) {
-        continue;
-      }
-      for (const auto& [labels, metric] : metric_family) {
-        if (metric && metric->is_enabled()) {
-          dump_metric_value(f.get(), full_name, *metric, labels);
+    f->open_array_section("metrics");
+    return seastar::do_for_each(
+      seastar::smp::all_cpus().begin(),
+      seastar::smp::all_cpus().end(),
```

Comment:
Let's make this a reusable helper in smp_helpers.h by splitting sharded_map_seq into sharded_map_seq and reactor_map_seq.  Something like:

```

/**
 * reactor_map_seq
 *
 * Invokes f on each reactor sequentially, Caller may assume that
 * f will not be invoked concurrently on multiple cores.
 */
template <typename F>
auto reactor_map_seq(F &&f) {
  using ret_type = decltype(f());
  if constexpr (is_errorated_future_v<ret_type>) {
    auto ret = crimson::do_for_each(
      seastar::smp::all_cpus().begin(),
      seastar::smp::all_cpus().end(),
      [f=std::move(f)](auto core) mutable {
	return seastar::smp::submit_to(
	  core,
	  [&f] {
	    return std::invoke(f);
	  });
      });
    return ret_type(ret);
  } else {
    return seastar::do_for_each(
      seastar::smp::all_cpus().begin(),
      seastar::smp::all_cpus().end(),
      [f=std::move(f)](auto core) mutable {
	return seastar::smp::submit_to(
	  core,
	  [&f] {
	    return std::invoke(f);
	  });
      });
  }
}

/**
 * sharded_map_seq
 *
 * Invokes f on each shard of t sequentially.  Caller may assume that
 * f will not be invoked concurrently on multiple cores.
 */
template <typename T, typename F>
auto sharded_map_seq(T &t, F &&f) {
  return reactor_map_seq(
    [&t, f=std::forward<F>(f)]() mutable {
      return std::invoke(f, t.local());
    });
}
```